Gems/AtomTressFX/Assets/Shaders/HairSimulationCompute.azsl

```diff
@@ -0,0 +1,437 @@
+                pos_plus_one.xyz += del.xyz;
+
+
+            g_HairVertexPositions[globalVertexIndex].xyz = pos.xyz;
+            g_HairVertexPositions[globalVertexIndex + 1].xyz = pos_plus_one.xyz;
+        }
+    }
+}
+
+//--------------------------------------------------------------------------------------
+//
+// LengthConstriantsWindAndCollision
+//
+// Compute shader to move the vertex position based on wind, maintain the lenght constraints
+// and handles collisions.
+//
+// One thread computes one vertex.
+//
+//--------------------------------------------------------------------------------------
+[numthreads(THREAD_GROUP_SIZE, 1, 1)]
+void LengthConstriantsWindAndCollision(uint GIndex : SV_GroupIndex,
+                  uint3 GId : SV_GroupID,
+                  uint3 DTid : SV_DispatchThreadID)
+{
+    uint globalStrandIndex, localStrandIndex, globalVertexIndex, localVertexIndex, numVerticesInTheStrand, indexForSharedMem, strandType;
+    CalcIndicesInVertexLevelMaster(GIndex, GId.x, globalStrandIndex, localStrandIndex, globalVertexIndex, localVertexIndex, numVerticesInTheStrand, indexForSharedMem, strandType);
+
+    uint numOfStrandsPerThreadGroup = g_NumOfStrandsPerThreadGroup;
+
+    //------------------------------
+    // Copy data into shared memory
+    //------------------------------
+    sharedPos[indexForSharedMem] = g_HairVertexPositions[globalVertexIndex];
+    sharedLength[indexForSharedMem] = g_HairRestLengthSRV[globalVertexIndex] * CM_TO_METERS;
+
+    GroupMemoryBarrierWithGroupSync();
+
+/*
+    //------------
+    // Wind
+    //------------
+    if ( g_Wind.x != 0 || g_Wind.y != 0 || g_Wind.z != 0 )
+    {
+        float4 force = float4(0, 0, 0, 0);
+
+        float frame = g_Wind.w;
+
+        if ( localVertexIndex >= 2 && localVertexIndex < numVerticesInTheStrand-1 )
+        {
+            // combining four winds.
+            float a = ((float)(globalStrandIndex % 20))/20.0f;
+            float3  w =  a* g_Wind.xyz + (1.0f - a) * g_Wind1.xyz + a * g_Wind2.xyz + (1.0f - a) * g_Wind3.xyz;
+//            float3  w = float3(5.2, 0, 0);
+
+            uint sharedIndex = localVertexIndex * numOfStrandsPerThreadGroup + localStrandIndex;
+
+            float3 v = sharedPos[sharedIndex].xyz - sharedPos[sharedIndex+numOfStrandsPerThreadGroup].xyz;
+            float3 force = -cross(cross(v, w), v);
+            sharedPos[sharedIndex].xyz += force*g_TimeStep*g_TimeStep;
+        }
+    }
+
+    GroupMemoryBarrierWithGroupSync();
+*/
+    //----------------------------
+    // Enforce length constraints
+    //----------------------------
+    uint a = floor(numVerticesInTheStrand/2.0f);
+    uint b = floor((numVerticesInTheStrand-1)/2.0f);
+
+    int nLengthContraintIterations = GetLengthConstraintIterations();
+
+    for ( int iterationE=0; iterationE < nLengthContraintIterations; iterationE++ )
+    {
+        uint sharedIndex = 2*localVertexIndex * numOfStrandsPerThreadGroup + localStrandIndex;
```

Comment:
this is because before it is not a real vertex but a guide point along the curve.  It is then going through expansion with maximum facing towards the viewer tin order to create right and left matching vertices.
I'll throw in a remark about it.